### DISCUSSION TITLE
github actions job timeout

### DISCUSSION THREAD

**Discussion by VijayReddyTR** [community/community#108006] [Actions]
### Select Topic Area

General

### Body

I am trying to check how much time the job runs in a selfhosted runner and i have used below code to test it.

jobs:
  time-check:
    runs-on: self_runner
    timeout-minutes: 600
    steps:
      - name: Check time every hour
        run: |
          $maxHours = 10 
          $counter = 0
          while ($counter -lt $maxHours) {
            # Print current date and time
            $currentDateTime = Get-Date -Format "yyyy-MM-dd HH:mm:ss"
            Write-Output "Current time: $currentDateTime"
            # Wait for an hour
            Start-Sleep -Seconds 3600
            $counter++
          }

I think it ran for 6 hrs and then after that it went to queue. Even after 10 hrs it keeps running. It says as below:
Waiting for a runner to pick up this job...

What should i do so that i can run the job for more than 6 hrs?

**Comment by Pranith-Kanchan** [comment]
To extend the runtime of your job beyond the default 6-hour timeout limit in GitHub Actions, you'll need to implement strategies to work around this constraint. One approach is to break down your task into smaller, more manageable units that can each complete within the 6-hour limit. This approach, known as job chaining, involves splitting your job into sequential steps or parallel processes that can be executed independently. By structuring your workflow in this way, you can ensure that each individual job completes within the time limit while achieving the overall task's objectives.

Additionally, consider optimizing your code and processes to reduce execution time. Analyze the operations performed within your job to identify potential bottlenecks or areas for improvement. Optimizing algorithms, minimizing unnecessary computations, and parallelizing certain tasks can help streamline execution and reduce overall runtime. Furthermore, consider leveraging parallelism where applicable to distribute workload across multiple resources simultaneously, thereby maximizing efficiency and reducing the time required to complete the job.

If your task still exceeds the 6-hour limit even after optimization, you may need to explore alternative solutions such as dedicated infrastructure. Running your job on dedicated servers or cloud instances outside of GitHub Actions can provide greater flexibility in terms of runtime limits and resource allocation. However, this approach typically requires additional setup, maintenance, and potentially increased costs compared to utilizing GitHub Actions.

Ultimately, the choice of approach depends on the specific requirements and constraints of your task, as well as the available resources and infrastructure. By carefully analyzing your workflow, optimizing code and processes, and considering alternative execution environments, you can overcome the 6-hour timeout limitation and successfully complete your job within the desired timeframe.

**Comment by VijayReddyTR** [comment]
I have a self hosted runner on our VM. The job running for 10 hrs as per timeout mentioned. But the job is not visible(running in background) after 6 hrs and saying waiting the runner to pickup...
